grid_map_rviz_plugin/include/grid_map_rviz_plugin/GridMapDisplay.hpp

```diff
@@ -6,74 +6,91 @@
  *  Institute: ETH Zurich, ANYbotics
  */
 
-#pragma once
+#ifndef GRID_MAP_RVIZ_PLUGIN__GRIDMAPDISPLAY_HPP_
+#define GRID_MAP_RVIZ_PLUGIN__GRIDMAPDISPLAY_HPP_
 
-#ifndef Q_MOC_RUN
+// #ifndef Q_MOC_RUN
 #include <grid_map_ros/grid_map_ros.hpp>
-#include <grid_map_msgs/GridMap.h>
+#include <grid_map_msgs/msg/grid_map.hpp>
+
+#include <rviz_common/message_filter_display.hpp>
+#include <rviz_common/properties/float_property.hpp>
+#include <rviz_common/properties/bool_property.hpp>
+#include <rviz_common/properties/int_property.hpp>
+#include <rviz_common/properties/color_property.hpp>
+#include <rviz_common/properties/enum_property.hpp>
+#include <rviz_common/properties/editable_enum_property.hpp>
+
 #include <boost/circular_buffer.hpp>
-#include "grid_map_rviz_plugin/modified/message_filter_display.h"
-#endif
+// #endif
 
-namespace Ogre {
+namespace Ogre
+{
 class SceneNode;
 }
 
-namespace rviz {
+namespace rviz_common
+{
+namespace properties
+{
 class BoolProperty;
 class ColorProperty;
 class FloatProperty;
 class IntProperty;
 class EnumProperty;
 class EditableEnumProperty;
-}
+}  // namespace properties
+}  // namespace rviz_common
 
-namespace grid_map_rviz_plugin {
+namespace grid_map_rviz_plugin
+{
 
 class GridMapVisual;
-class GridMapDisplay : public MessageFilterDisplayMod<grid_map_msgs::GridMap>
+class GridMapDisplay : public rviz_common::MessageFilterDisplay<grid_map_msgs::msg::GridMap>
 {
-Q_OBJECT
- public:
+  Q_OBJECT
+
+public:
   GridMapDisplay();
   virtual ~GridMapDisplay();
 
- protected:
-  virtual void onInitialize();
+protected:
+  void onInitialize() override;
 
-  virtual void reset();
+  void reset() override;
 
- private Q_SLOTS:
+private Q_SLOTS:
   void updateHistoryLength();
   void updateHeightMode();
   void updateColorMode();
   void updateUseRainbow();
   void updateAutocomputeIntensityBounds();
   void updateVisualization();
 
- private:
+private:
   // Callback for incoming ROS messages
-  void processMessage(const grid_map_msgs::GridMap::ConstPtr& msg);
+  void processMessage(grid_map_msgs::msg::GridMap::ConstSharedPtr msg) override;
```

Comment:
Is the const intentionally removed?

Reply:
This is the [method](https://github.com/ros2/rviz/blob/d59072192205a92acd9bc84605420a815c885d00/rviz_common/include/rviz_common/message_filter_display.hpp#L214) being overridden. Also here is an [example](https://github.com/ros2/rviz/blob/d59072192205a92acd9bc84605420a815c885d00/rviz_default_plugins/src/rviz_default_plugins/displays/camera/camera_display.cpp#L568) of it being used.

I tried to stick to the same parameter type as in the virtual function above and assumed `::ConstSharedPtr` has the const part already. Having said that, I just tried adding const back and it is also building fine, not sure if there is a need to have it back though.